samples/go/json-import/json_importer.go

```diff
@@ -8,16 +8,68 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
 	"log"
+	"math/big"
 	"net/http"
 	"os"
+	"time"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/spec"
 	"github.com/attic-labs/noms/go/util/jsontonoms"
+	"github.com/dustin/go-humanize"
 	flag "github.com/juju/gnuflag"
 )
 
+func newProgressReader(r io.Reader) io.Reader {
```

Comment:
we have `https://github.com/attic-labs/noms/go/util/progressreader` too.  Though it would force this to be written in a more callbacky style, I think overall it would be better to share code?